duckduckgo/Android
# Tabs

Reviewer: @subsymbolic 
Asana: https://app.asana.com/0/0/350058423386223
CC:

**Description**: Implement basic tabs
Known bad behavior: when closing all tabs in the tab switcher the tab automatically creates a new tab, this will be fixed in the Navigation Task (https://app.asana.com/0/0/361925093999571)


**Steps to test this PR**:
Test 1
1. Use the tab switcher to create new tabs and switch
2. Confirm that tabs are preserved when switching back

Test 2
1. Create a few tabs with some log in.
2. Click the forget button in the tab switcher
3. Access the previous URL with log in and see that the credentials are erased

Test 3
1. Create a few tabs
2. Make the system kill the process
3. Open the app and confirm that the session is restored

Test 4
1. Create a few tabs
2. Swipe the app from the system app switcher
3. Open the app and confirm that the session is not restored


###### Reviewer Checklist:


###### PR DRI Checklist:
